### PR TITLE
cpu/stm32l1: header updated and isr vector cleanup

### PR DESCRIPTION
**Pull request by haukepetersen**
follow up on #7535

This PR also updates the vendor headers as the used one was quite outdated.

### CODE REVIEW

**Comment by haukepetersen** [comment]
rebased

**Comment by aabadie** [comment]
needs rebase

**Comment by haukepetersen** [comment]
rebased.

**Comment by haukepetersen** [comment]
not good. Will check.

**Comment by haukepetersen** [comment]
I have a different behavior here: `tests/periph_gpio` and `tests/i2c` work as expected, but `default` goes into hard-fault once I input something to the shell...

**Comment by smlng** [comment]
mhm, I just made one pass, so I was lucky on some - but in general its not deterministic?

**Comment by smlng** [comment]
@haukepetersen: yep now `example/defaults` failed for me, too 😢

**Comment by haukepetersen** [comment]
To simplify things, I am looking into `tests/shell` right now. As the shell works with the `tests/periph_gpio` test, I con't expect this too be a problem with the uart driver, but rather something more system related.

What I can tell so far: `tests/shell` fails with this PR (hard fault on rx), but if I comment out the `DISABLE_MODULE += auto_init` in the Makefile, then it works... But no idea what this means so far.

**Comment by haukepetersen** [comment]
seems like we broke the board/cpu a while back. git bisect lead me to believe that `dd49f225321d3929f41c6108749c04eab0eb9571` broke the board, though I can't really tell why this is.

**Comment by smlng** [comment]
but that commit would break then all cortex-m based CPUs which it certainly doesn't because others still work.

**Comment by haukepetersen** [comment]
Just tried it: manuall reverting the changes from dd49f225321d3929f41c6108749c04eab0eb9571 on the current master makes the nucleo-l152 behave as expected again. So something is with this change that the l1 doesn't like.

**Comment by smlng** [comment]
@haukepetersen the difference is that `irq_restore` is mapped to `irq_arch_restore` which is 
```
/**
 * @brief Restore the state of the IRQ flags
 */
void irq_arch_restore(unsigned int state)
{
    __set_PRIMASK(state);
}
```

problem: it doesn't call `__enable_irq()` while `irq_disable()` is mapped to `irq_arch_disable` which is:
```
/**
 * @brief Disable all maskable interrupts
 */
unsigned int irq_arch_disable(void)
{
    uint32_t mask = __get_PRIMASK();
    __disable_irq();
    return mask;
}
```

and does `__disable_irq()`

**Comment by smlng** [comment]
we can say for sure, that replacing these to lines does not result in the same functions called, at least the now used restore does something different and does not call `__enable_irq()`, I wonder why other platforms are not showing this behaviour. I mean I mostly tests with `remote-revb` and PhyNode (`pba-d-01-kwx2`) which are both cortex-m as well, but so far no problem with them.

**Comment by smlng** [comment]
> we can say for sure, that replacing these to lines 

as in the commit by @vincent-d  you found

**Comment by smlng** [comment]
ah forget about the `remotes`, they do not use `cortexm_sleep` so they shouldn't be affected, but maybe its a similar problem why I had to exclude them.

**Comment by haukepetersen** [comment]
Just put the question into #7385. Either there is a quick fix, or this will be one of those hard to fix issues...

Anyway, I am pretty sure the issue at hand is unrelated to the changes in this PR. Would you be ok with merging this PR and keeping track of the broken `stm32l1` in a different place?

**Comment by aabadie** [comment]
This needs rebase @haukepetersen.

Maybe @kYc0o you can give this one a try (since you seem to also have the hardware) ?

File: cpu/stm32l1/vectors.c [hunk 1]
```diff
@@ -29,130 +28,124 @@ void dummy_handler(void) {
...
-    isr_usart1,             /* [37] USART1 global Interrupt */
-    isr_usart2,             /* [38] USART2 global Interrupt */
-    isr_usart3,             /* [39] USART3 global Interrupt */
-    isr_exti,               /* [40] External Line[15:10] Interrupts */
-    isr_rtc_alarm,          /* [41] RTC Alarm through EXTI Line Interrupt */
-    isr_usb_fs_wkup,        /* [42] USB FS WakeUp from suspend through EXTI Line Interrupt */
-    isr_tim6,               /* [43] TIM6 global Interrupt */
-    isr_tim7,               /* [44] TIM7 global Interrupt */
+    /* shared vectors for all family members */
+    [WWDG_IRQn               ] = isr_wwdg,                 /* [ 0] Window WatchDog Interrupt */
```
Comment (kYc0o) [inline cpu/stm32l1/vectors.c]: Any special reason to align the closing bracket with so much spaces?

File: cpu/stm32l1/vectors.c [hunk 2]
```diff
@@ -29,130 +28,124 @@ void dummy_handler(void) {
...
+    [SPI2_IRQn               ] = isr_spi2,                 /* [36] SPI2 global Interrupt */
+    [USART1_IRQn             ] = isr_usart1,               /* [37] USART1 global Interrupt */
+    [USART2_IRQn             ] = isr_usart2,               /* [38] USART2 global Interrupt */
+    [USART3_IRQn             ] = isr_usart3,               /* [39] USART3 global Interrupt */
+    [EXTI15_10_IRQn          ] = isr_exti,                 /* [40] External Line[15:10] Interrupts */
+    [RTC_Alarm_IRQn          ] = isr_rtc_alarm,            /* [41] RTC Alarm through EXTI Line Interrupt */
+    [USB_FS_WKUP_IRQn        ] = isr_usb_fs_wkup,          /* [42] USB FS WakeUp from suspend through EXTI Line Interrupt */
+    [TIM6_IRQn               ] = isr_tim6,                 /* [43] TIM6 global Interrupt */
+    [TIM7_IRQn               ] = isr_tim7,                 /* [44] TIM7 global Interrupt */
+
```
Comment (kYc0o) [inline cpu/stm32l1/vectors.c]: Any special reason fo the extra space here and below? (just curiosity, I'm not against it)

**Comment by smlng** [comment]
needs rebase

**Comment by smlng** [comment]
as such changes are done for other MCU already, this should be moved forward (again). Hence, don't close.

**Pull request closed without merging**